src/main/java/com/company/managementservice/controller/OrganisationController.java

```diff
@@ -0,0 +1,116 @@
+            String errMsg = "";
+            for (FieldError err: bindingResult.getFieldErrors()) {
+                errMsg += err.getField() + " is " + err.getCode();
+            }
+            throw new MethodArgumentNotValidException(errMsg);
+        }
+        return new ServiceResponse<>(
+                new BaseMessageResponse<>(
+                        organisationService.saveOrganisation(organisationDto),
+                        HttpStatus.OK, true));
+
+    }
+
+    @GetMapping(value = "/{organisationId}/details")
+    public ResponseEntity<?> getOrganisation(@PathVariable @NonNull Integer organisationId) throws NotFoundException {
+        log.info("OrganisationController : getOrganisationDetails : Received Request to get Organisation Details :{}",
+                 organisationId);
+        return new ServiceResponse<>(
+                new BaseMessageResponse<>(
+                        organisationService.getOrganisation(organisationId), HttpStatus.OK, true));
+
+    }
+
+    @PutMapping("/{organisationId}/update-details")
+    public ResponseEntity<?> updateOrganisation(@Valid @RequestBody OrganisationDto organisationDto,
+                                                BindingResult bindingResult,
+                                                @PathVariable Integer organisationId)
+            throws NotFoundException, MethodArgumentNotValidException {
+        log.info("OrganisationController : PutOrganisationDetails : Received Request to put Organisation Details  :{}",
+                 organisationId);
+        if (bindingResult.hasErrors()) {
+            String errMsg = "";
+            for (FieldError err: bindingResult.getFieldErrors()) {
+                errMsg += err.getField() + " is " + err.getCode();
+            }
+            throw new MethodArgumentNotValidException(errMsg);
+        }
+        return new ServiceResponse<>(
+                new BaseMessageResponse<>(organisationService.updateOrganisation(organisationDto, organisationId),
+                                          HttpStatus.OK, true));
+    }
+
+    @DeleteMapping("/{organisationId}/remove-department/{departmentId}")
+    public ResponseEntity<?> removedepartment(@NonNull @PathVariable Integer organisationId,
+                                              @NonNull @PathVariable Long departmentId) throws NotFoundException {
+        log.info(
+                "OrganisationController : removeDepartment : Received request to remove Department from organisation id :{} ",
+                organisationId);
+        organisationService.removeDepartment(organisationId, departmentId);
+        return new ServiceResponse<BaseMessageResponse>(
+                new BaseMessageResponse(
+                        "Updated Successfully ",
+                        HttpStatus.OK, true));
+    }
+
+    @DeleteMapping("/{id}/remove-organisation")
+    public ResponseEntity<?> removeorganisation(@NonNull @PathVariable Integer id) throws NotFoundException {
```

Comment:
lets loose the verb after id. we are not working on any nested resource here under id, right?

Reply:
most of the restful apis have actions as query param if needed.